test/gtest/uct/ib/test_ib.cc

```diff
@@ -376,6 +376,59 @@ UCS_TEST_P(test_uct_ib_gid_idx, non_default_gid_idx, "GID_INDEX=1") {
 UCT_INSTANTIATE_IB_TEST_CASE(test_uct_ib_gid_idx);
 
 
+#if HAVE_DEVX
+class test_uct_ib_sl : public test_uct_ib_with_specific_port {
+public:
+    void init() {
+        test_uct_ib_with_specific_port::init();
+    }
+
+    void cleanup() {
+        test_uct_ib_with_specific_port::cleanup();
+    }
+
+    void check_port_attr() {
+        ucs_status_t status;
+        ucs::handle<uct_md_h> uct_md;
+
+        UCS_TEST_CREATE_HANDLE(uct_md_h, uct_md, uct_ib_md_close, uct_ib_md_open,
+                               &uct_ib_component,
+                               ibv_get_device_name(m_ibctx->device),
+                               m_md_config);
+
+        uct_ib_mlx5_md_t *ib_md = ucs_derived_of(uct_md, uct_ib_mlx5_md_t);
+
+        /* check if OOO SL mask can be detected for the port */
+        status = uct_ib_mlx5_devx_query_ooo_sl_mask(ib_md, m_port,
+                                                    &m_ooo_sl_mask);
+        EXPECT_TRUE((status == UCS_OK) || (status == UCS_ERR_UNSUPPORTED));
+        if (status != UCS_OK) {
+            UCS_TEST_SKIP_R("OOO SL mask couldn't be detected for " +
+                            m_dev_name + ":" + ucs::to_string(m_port));
+        }
+    }
+
+protected:
+    uint16_t m_ooo_sl_mask;
+};
+
+UCS_TEST_P(test_uct_ib_sl, check_ib_sl_config) {
+    for (uint8_t sl = 0; sl < UCT_IB_SL_NUM; ++sl)  {
+        if (!has_transport("rc_verbs") && !has_transport("ud_verbs")) {
+            modify_config("IB_AR_ENABLE",
+                          (m_ooo_sl_mask & UCS_BIT(sl)) ? "y" : "n");
+        }
+        modify_config("IB_SL", ucs::to_string(static_cast<uint16_t>(sl)));
+
+        test_uct_ib::init();
+        send_recv_short();
+        test_uct_ib::cleanup();
+    }
+}
+
+UCT_INSTANTIATE_IB_TEST_CASE(test_uct_ib_sl);
```

Comment:
can we instantiate only on mlx5 transports, but not change the common UCT_INSTANTIATE_IB_TEST_CASE macro?

Reply:
let's test that we could select different SLs for Verbs transports and everything is ok?

Reply:
ok